lms/views/lti/basic_launch.py

```diff
@@ -112,24 +112,40 @@ def reconfigure_assignment_config(self):
         schema=ConfigureAssignmentSchema,
         renderer="lms:templates/lti/basic_launch/basic_launch.html.jinja2",
     )
-    def configure_assignment(self):
+    def configure_assignment_callback(self):
         """
         Save the configuration from the file-picker for future launches.
 
         We then continue as if we were already configured with this document
         and display it to the user.
         """
-        extra = {}
-        if group_set := self.request.parsed_params.get("group_set"):
-            extra["group_set_id"] = group_set
-
-        # Make any product-specific actions after configuring the assignment
-        self.request.product.plugin.misc.post_configure_assignment(self.request)
+        return self._configure_and_show_document()
 
-        return self._show_document(
-            document_url=self.request.parsed_params["document_url"],
-            assignment_extra=extra,
+    @view_config(
+        route_name="edit_assignment",
+        permission=Permissions.LTI_CONFIGURE_ASSIGNMENT,
+        schema=ConfigureAssignmentSchema,
+        renderer="lms:templates/lti/basic_launch/basic_launch.html.jinja2",
+    )
+    def edit_assignment_callback(self):
+        """Edit the configuration of an existing assignment."""
+        assignment = self.assignment_service.get_assignment(
+            tool_consumer_instance_guid=self.request.lti_params[
+                "tool_consumer_instance_guid"
+            ],
+            resource_link_id=self.request.lti_params.get("resource_link_id"),
         )
+        self.request.registry.notify(
+            LTIEvent(
+                request=self.request,
+                type=LTIEvent.Type.EDITED_ASSIGNMENT,
+                data={
+                    "old_url": assignment.document_url,
```

Comment:
This will be useful to track the usage and debug in the future.

If we do need this in a more structure manner in the future we can find a better design then.

I reckon this is better than just not recording this at all and :shrug:  when we might need it.